test/conftest.py

```diff
@@ -310,3 +311,7 @@ def pytest_report_collectionfinish(self) -> Optional[str]:
     def pytest_runtest_protocol(self, item, nextitem) -> None:
         self.lastrun = item.nodeid
         self.cache.set(self.directory, self.lastrun)
+
+    def pytest_sessionfinish(self, session, exitstatus):
```

Comment:
The function is a hook for pytest, called by pytest when the pytest session finishes.  I put it as a is part of a plugin for CI that keeps track of the last test that got run so that if you rerun you can skip all the previous tests that have already run.